pkg/kube/logs.go

```diff
@@ -32,7 +35,7 @@ func (r *logsReader) GetLogsByJobAndContainerName(ctx context.Context, job *batc
 		return nil, fmt.Errorf("getting pod controlled by job: %q: %w", job.Namespace+"/"+job.Name, err)
 	}
 	if pod == nil {
```

Comment:
@jw-s thank you for the contribution:
maybe we should handle both cases :
1. Job deleted from Etcd, but the job exist in controller cache
2. pod deleted from Etcd and Job exist in Etcd and cache (as appear in your PR)

Reply:
Hi @chen-keinan Point 1 is covered before it even gets to this stage, earlier on in the reconcile look it checks if the object exists for example: https://github.com/aquasecurity/starboard/blob/main/pkg/operator/controller/vulnerabilityreport.go#L97

Reply:
@jw-s to `reconcile loop` you mention is for the `workload`, I'm referring to the 2nd reconcile loop for `scan jobs`
where 1st we tried to get it from cache :
`log.V(1).Info("Getting job from cache")
		err := r.Client.Get(ctx, req.NamespacedName, job)
		if err != nil {
                                if errors.IsNotFound(err) {
				log.V(1).Info("Ignoring cached job that must have been deleted")
				return ctrl.Result{}, nil
			}`
and only afterwards we get tries to get the job from Etcd :
`
func (r *logsReader) getPodByJob(ctx context.Context, job *batchv1.Job) (*corev1.Pod, error) {
	refreshedJob, err := r.clientset.BatchV1().Jobs(job.Namespace).Get(ctx, job.Name, metav1.GetOptions{})
	`

Reply:
@jw-s It will be great if you could add the check for `Job not found`,  referring to other PR opened on same issue #950

Reply:
@chen-keinan The new commit checks if the job isn't found and returns nil causing the controller to no requeue.